### PR TITLE
adrgen: init at 0.4.0-beta

### LINKED ISSUE #152638: ADRgen
**ADRgen**
_A command-line tool for generating and managing Architecture Decision Records_

**Metadata**

* homepage URL: https://github.com/asiermarques/adrgen
* source URL: https://github.com/asiermarques/adrgen
* license: MIT
* platforms: unix

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Closes: #152638

Unit tests succeed, and the binary seems to function as expected from my basic testing.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by wraithm — APPROVED** [review]

File: pkgs/tools/misc/adrgen/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "sha256-2ZE/orsfwL59Io09c4yfXt2enVmpSM/QHlUMgyd9RYQ=";
+  };
+
+  vendorSha256 = "sha256-aDtUD+KKKSE0TpSi4+6HXSBMqF/TROZZhT0ox3a8Idk=";
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    $out/bin/adrgen version
+  '';
```
Comment (legendofmiracles) [inline pkgs/tools/misc/adrgen/default.nix:20]: This should rather be a package test https://nixos.org/manual/nixpkgs/stable/#ssec-inline-package-tests-writing

Comment (bobby285271) [reply pkgs/tools/misc/adrgen/default.nix:20]: And there is [`testVersion`](https://github.com/NixOS/nixpkgs/blob/de49d559382ea05a304589b6fc13c9be68351810/pkgs/build-support/trivial-builders.nix#L724-L755) if you are interested in that. See https://github.com/NixOS/nixpkgs/commit/255c2849248e331d7f0e56a041b80e0014a19177 for how to use it.

Comment (IvarWithoutBones) [reply pkgs/tools/misc/adrgen/default.nix:20]: Thanks! I didn't know about these, I'll use `testVersion` for future packages as well :)

**Comment by IvarWithoutBones** [comment]
Sorry for taking a few days to respond, should be ready to merge now :+1:

File: pkgs/tools/misc/adrgen/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  vendorSha256 = "sha256-aDtUD+KKKSE0TpSi4+6HXSBMqF/TROZZhT0ox3a8Idk=";
+
+  passthru.tests.version = testVersion {
+    package = adrgen;
+    command = "adrgen version";
+    version = "v${version}";
+  };
+
+  meta = with lib; {
```
Comment (legendofmiracles) [inline pkgs/tools/misc/adrgen/default.nix:27]: Missing description

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/adrgen/default.nix (hunk 2 above)
Comment (IvarWithoutBones) [reply pkgs/tools/misc/adrgen/default.nix:27]: Huh, I could've sworn added one. Should be good now tho :+1:

**Comment by wraithm** [comment]
Thank you very much @IvarWithoutBones!

**Pull request merged**